README.md

```diff
@@ -1,2 +1,95 @@
 # gnosis-labs-zuberlin2024
+
 Repository for the hackathon run by Gnosis Labs at ZuBerlin 2024.
+
+## Support
+
+Contact us at https://t.me/+Fb0trLKZdMw2MTQ8.
+
+## Setup
+
+Install the project dependencies with `poetry`, using Python 3.10 (you can use [pyenv](https://github.com/pyenv/pyenv) to manage multiple Python versions):
+
+```bash
+python3.10 -m pip install poetry
+python3.10 -m poetry install
+python3.10 -m poetry shell
+```
+
+Copy `.env.example` to `.env` and fill in the values:
+
+### OpenAI API key
+
+We will provide you with OpenAI key that's allowed to use gpt-3.5-turbo and embedding models, contact us on the TG group above.
+
+However, everyone is welcome to use arbitrary LLM if wanted.
+
+### Tavily API key
+
+Create a free acount on https://tavily.com and get the key there.
+
+Again, everyone is welcome to use arbitrary search engines, combine them, or even do a totally different approaches!
+
+### Private key on Gnosis Chain
+
+Use your existing or create a new wallet on Gnosis Chain. 
+
+By default the script will do only very tiny bets (0.00001 xDai per market), but of course, you can contact us on the TG group above with your public key to get some free xDai.
+
+## Task
+
+Your task is to modify `predict` function in `trader/prediction.py` by any means necessary.
+
+Goal of the `predict` function is, given an `question` about the future, answer it with either `True` (the answer is `yes`), `False` (if the answer is `no`) or `None` (if the prediction failed).
+
+All the questions are guaranteed to be about the future and to be in a binary yes/no format.
+
+You can play with the prompts, different approaches, different LLMs, search engines, or anything you can think of.
+
+The code can be messy, the only thing we ask you is for it to be reproducible on our machines, and to help with that, there is `mypy` as the only check of CI pipeline on Github.
+
```

Comment:
I was thinking of adding a deployment script to an AWS lambda or equiv GCP function, so that they can deploy themselves and we can keep track?
Otherwise we need to deploy several functions and errors might occur, what do you think?

Reply:
My idea was that they would execute the `main.py` script once and bet on all markets created by that public address I listed there—no need to keep it running afterwards.